File: 02-data-processing/csv.py
# Python Module csv

def mapToObj(item, attribs):
    codeItem = {}
    splited = item.split(';')
    index = 0
    for value in splited:
      codeItem[attribs[index]] = value
      index += 1

    return codeItem

def load(filePath):
    f = open(filePath, "r", encoding='utf-8')
    data = f.read()
    list = data.split('\n');
    
    attribs = list.pop(0).split(';')

    # print(mapToObj(list[0], attribs))
    objectList = [];
    for item in filter(lambda item: item, list):
        objectList.append(mapToObj(item, attribs))

    return objectList

File: 02-data-processing/csv.js
const fs = require('fs');

module.exports = {
    load: function(filePath) {
        const data = fs.readFileSync(filePath, 'utf8');
        const list = data.split('\n').map(item => item.trim());
        const attribs = list.shift().split(';');
        return list.filter(item => !!item).map(item => {
            const codeItem = {};
            item.split(';').forEach((value, index) => {
            codeItem[attribs[index]] = value;
            });
            return codeItem;
        });
    }
}
